### PR TITLE
PKI: Use ECDSA keys by default 

### PR DESCRIPTION
**Pull request by alvaroaleman**
These are computationally cheaper to generate and use and more secure
and smaller. They are also FIPS-compliant[0].
Compatibility with exiting RSA CAs is retained.
    
Due to lacking fips-compliance we can not use ed25519, as of time of
writing they are only proposed for that: [1].
    
For comparison:
    
Previously:
    
 ```
go test -race -count=5 ./control-plane-operator/controllers/hostedcontrolplane/pki/
ok github.com/openshift/hypershift/control-plane-operator/controllers/hostedcontrolplane/pki 25.986s
 ```
    
With this change:
    
```
go test -race -count=5 ./control-plane-operator/controllers/hostedcontrolplane/pki/
ok      github.com/openshift/hypershift/control-plane-operator/controllers/hostedcontrolplane/pki       6.939s
```
    
[0]: https://csrc.nist.gov/publications/detail/fips/186/4/final
[1]: https://csrc.nist.gov/publications/detail/fips/186/5/draft

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, use `fixes #<issue_number>(, fixes #<issue_number>, ...)` format, where issue_number might be a GitHub issue, or a Jira story*:
Fixes #

**Checklist**
- [ ] Subject and description added to both, commit and PR.
- [ ] Relevant issues have been referenced.
- [ ] This change includes docs. 
- [ ] This change includes unit tests.

### CODE REVIEW

File: support/certs/tls.go [hunk 1]
```diff
@@ -228,27 +234,47 @@ func CSRToPem(cert *x509.CertificateRequest) []byte {
...
-func PublicKeyToPem(key *rsa.PublicKey) ([]byte, error) {
+func PublicKeyToPem(key crypto.PublicKey) ([]byte, error) {
 	keyInBytes, err := x509.MarshalPKIXPublicKey(key)
 	if err != nil {
 		return nil, errors.Wrap(err, "failed to MarshalPKIXPublicKey")
 	}
 	keyinPem := pem.EncodeToMemory(
 		&pem.Block{
-			Type:  "RSA PUBLIC KEY",
+			Type:  "PUBLIC KEY",
```
Comment (alvaroaleman) [inline support/certs/tls.go:244]: Using `RSA PUBLIC KEY` here was wrong: https://cs.opensource.google/go/go/+/refs/tags/go1.18:src/crypto/x509/x509.go;l=116

**Comment by sjenning** [comment]
Is this FIPS approved?

**Comment by alvaroaleman** [comment]
> Is this FIPS approved?

No,it is currently only proposedm seems we can not have nice things :( Ironically, FIPS allows DSA though. I've updated the PR and description to use ECDSA keys, they are an order of magnitude slower than ED25519 but still more than three times faster than RSA keys.

File: .gitleaks.toml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+[allowlist]
```
Comment (sjenning) [inline .gitleaks.toml:1]: what is this file for?

File: control-plane-operator/controllers/hostedcontrolplane/pki/cert_test.go [hunk 3]
```diff
@@ -12,8 +13,69 @@ import (
...
+YsdLDATHICIZMB8GA1UdIwQYMBaAFEI1WTEJDpk939vrYsdLDATHICIZMA8GA1Ud
+EwEB/wQFMAMBAf8wDQYJKoZIhvcNAQELBQADggEBAIZT6bIp4RhsZLjC0p7LSH4v
+aN+2JZKEMmCf/xgbJPizaZOvfyMykHIumGFXQmV1dser/pg3o5uAd9V6OhLXib6u
+lhyx0tWhcEi2WkMME/TMdwWKKt7RpnMPp1Aq8ZDCMArVWsqhYbjRFR7+Mz/+U46Y
+3Tl3wkWi+598ruiSKOZodwk1Nl0Sc3zR9/GphnmwKmXFukgdTv4HtGTjA8c3s+hB
+50WT43ndKOQLWaBkWhvev2vBz3K3ZAi1ENoDjFYwtLym1AlisLlKOOUeC6ChpLAr
+NakE7TNr5654KIzEGFHLA3h4UxNl33/m5zFgL0ByZTtCA1cvAnlLGOn7/MgN0Y4=
+-----END CERTIFICATE-----
+`),
+			CASignerKeyMapKey: []byte(`-----BEGIN RSA PRIVATE KEY-----
```
Comment (sjenning) [inline control-plane-operator/controllers/hostedcontrolplane/pki/cert_test.go:47]: should this be RSA?

File: .gitleaks.toml (hunk 2 above)
Comment (alvaroaleman) [reply .gitleaks.toml:1]: For posterity, redhat has a scanner that checks content that gets checked in for secrets and it found the RSA key I put into the _test.go file. This file adds an exception for that.

**Pull request merged**